src/praisonai-agents/praisonaiagents/agents/agents.py

```
@@ -1946,38 +1946,88 @@ def append_to_state(self, key: str, value: Any, max_length: Optional[int] = None
+        legacy semantic-recall lookups still find the state when memory is on.
+        """
+        state_data = self._team_state_payload(session_id)
+
+        # Primary, durable path: SessionStore (locked, atomic, memory-independent).
+        try:
+            from ..session.store import get_default_session_store
+            get_default_session_store().update_session_metadata(
+                session_id, team_session_state=state_data
             )
-    
+        except Exception as e:  # never fail a completed run on save
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)